// components/search/index.js
import React, {Component} from "react";
import {connect} from 'react-redux';

import {searchProduct} from 'actions';


class Search extends Component{
    constructor(props) {
        super(props);
        this.state = {
            isActive: false,
            value:''

        };
        //this.handleSubmit = this.handleSubmit.bind(this);
        //this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        console.log(event.target.value);
        this.setState({
            value: event.target.value
        })
    }

    handleSubmit(event) {
        console.log(this.state.value);
        event.preventDefault();
        this.props.searchProduct(this.state.value);
        const search = document.querySelector('.searchWindow');
        search.setAttribute('hidden', 'true');
        this.state.value = '';
    }

    toggleActive(){
        this.setState ({isActive: !this.state.isActive});
    }

    runScript = event => {
        if(event.keyCode === 13){
            return true;
        }
        return false;
    };

    renderSearch(flag){
        let body = document.getElementById('body');
        if(!flag) {
            body.classList.remove('modal-open');
            return '';
        }
        body.classList.add('modal-open');


        return (
            <div className='searchWindow'>
              <div className="modal fade in" id="myModal" role="dialog" aria-hidden='false' style={{display: 'block', paddingLeft: 0 + 'px'}} >
                <div className="modal-dialog">
                    <div className="modal-content">
                        <div className="modal-header">
                            <button type="button" className="close" data-dismiss="modal" onClick={this.toggleActive.bind(this)}>&times;</button>
                            <h4 className="modal-title">SEARCH HERE</h4>
                        </div>
                        <div className="modal-body">
                            <div className="input-group" >
                                <form method="get" className="searchform"  role="search" onSubmit={this.runScript && this.handleSubmit.bind(this)}>
                                    <input type="hidden" name="type" value="product"/>
                                    <input type="text" name="q" className="form-control control-search" defaultValue='' onChange={this.handleChange.bind(this)}/>
                                    <span className="input-group-btn">
                              <button className="btn btn-default button_search" type="button" onClick={this.handleSubmit.bind(this)}>
                                  <i data-toggle="dropdown"
                                     className="fa fa-search">
                                  </i>
                              </button>
                            </span>
                                </form>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
                <div className='modal-backdrop fade in'></div>
            </div>
        )
    }

    render() {
        let search;
        let flag = this.state.isActive;
        search = this.renderSearch(flag);

        return (
            <div>
                {search}
               <div className="topbar-search">

                <div className="search-popup dropdown" data-toggle="modal" data-target="#myModal">
                    <a href="#"><i className="icon-magnifier f-15" onClick={this.toggleActive.bind(this)}></i></a>
                </div>
               </div>

            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        isActive: state
    }
};

const mapDispatchToProps = {
    searchProduct
}

export default connect(mapStateToProps, mapDispatchToProps)(Search);
